src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final FruitDao fruitDao;
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(FruitDao fruitDao, OperationStrategy operationStrategy) {
+        this.fruitDao = fruitDao;
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            Fruit fruit = fruitDao.getFruitByName(transaction.getFruit());
+            OperationHandler handler = operationStrategy.get(transaction.getOperation());
```

Comment:
You should not directly get the handler from the strategy within the loop. Instead, consider refactoring the code to follow the Strategy pattern more closely by having a service that handles the strategy logic.

Reply:
--